pkg/util/logging/log_counter_hook.go

```diff
@@ -47,7 +53,31 @@ func (h *LogCounterHook) Fire(entry *logrus.Entry) error {
 	defer h.mu.Unlock()
 
 	h.counts[entry.Level]++
+	if h.entries[entry.Level] == nil {
+		h.entries[entry.Level] = &results.Result{}
+	}
+
+	namespace, isNamespacePresent := entry.Data["namespace"]
+	errorField, isErrorFieldPresent := entry.Data["error"]
+	resourceField, isResourceFieldPresent := entry.Data["resource"]
+	nameField, isNameFieldPresent := entry.Data["name"]
+
+	entryMessage := ""
+	if isResourceFieldPresent {
+		entryMessage = fmt.Sprintf("%s resource: /%s", entryMessage, resourceField.(string))
+	}
+	if isNameFieldPresent {
+		entryMessage = fmt.Sprintf("%s name: /%s", entryMessage, nameField.(string))
+	}
+	if isErrorFieldPresent {
+		entryMessage = fmt.Sprintf("%s error: /%s", entryMessage, errorField.(error).Error())
+	}
 
+	if isNamespacePresent {
```

Comment:
Namespace field is present in all logs emerging from "Item" related operations. The logger itself initiated with namespace  - example: https://github.com/vmware-tanzu/velero/blob/7139daf07a76de7bcc64f33d83f12315de4033f9/pkg/backup/item_backupper.go#L86

So any other error that happens outside the core backup flow has to be attributed to velero errors. 
Do you have any other scenarios in mind?

We are only checking for existence of Namespace field. Not if it is empty string or actual value.

Reply:
Adding generic "message" field will change BackupResults' behavior from Restore Results.
I didn't want to diverge them and keep them the same contract wise and logic wise as much as possible.

Reply:
@anshulahuja98 "Namespace field is present in all logs emerging from "Item" related operations." -- what about cluster-scoped resources? So the `Result` type on restore (where we identify errors as they're created, so we don't have this issue), currently has 3 buckets of information: Velero errors (not associated with namespace or item), Cluster errors (not associated with namespace, but associated with item) and Namespaces (associated with both namespace and item). If we can distinguish cluster-scoped item errors from non-item errors, then the Result struct still works for this.

Reply:
In backup flow (this PR) the data in result structs are populated via parsing the log, so it's hard to differentiate the velero level errors and Cluster Resource level errors, unless we update the code in the backup controller and let it explicitly add errors into Result struct.

Reply:
@reasonerjt Right. On the restore side, all of the funcs are designed to pass that Result struct back, so we keep a running list of errors and warnings, separated by namespace, cluster-scope, and non-resource errors. The best way to implement this would be to do the same for backup, but that would be a much larger bit of work. It might be worth looking at all of the places we create error/warning messages and see whether there's some way to differentiate in the error string (or in the logged values) which is which, but I haven't looked at the warnings and errors logged in detail recently, so I'm not sure this is possible.

Reply:
Per discussion in the community meeting today, we wanna continue with the approach to populate `result.json` for a backup via the log counter hook.  The risk for inaccuracy is low and having the result.json uploaded for a backup is a good improvement.

It seems when the backupper is handling a cluster-level resource, in the logger, the `namespace` field is present but is an empty string.  Like:
```
....
time="2023-02-15T12:04:29Z" level=info msg="Processing item" backup=velero/all logSource="pkg/backup/backup.go:348" name=endpoint-controller namespace= progress= resource=flowschemas.flowcontrol.apiserver.k8s.io
...
```

Another trivial comment, the `logCounterHook` may be better renamed to `logHook`?

@anshulahuja98 please let us know if you wanna make the change.

Once the change is done and the commits are squashed I think this PR can be merged.